lib/pcp/simple_logger.rb

```diff
@@ -0,0 +1,25 @@
+require 'logger'
+
+# This logger just adds events to the @events property, so when you
+# .inspect it you can see what was logged.
+module PCP
+  class SimpleLogger < ::Logger
+    def initialize(*args)
+      @events = []
+    end
+
+    def add(severity, message = nil, progname = nil)
+      if message.nil?
+        if block_given?
+          message = yield
+        else
+          message = progname
+        end
+      end
+
+      @events << {:when => Time.now.to_f,
```

Comment:
It would, but it will lose us the fractional part of the timestamp which I think is more important in this logging.

```
$ ruby -rtime -e 't = Time.now ; puts [t.iso8601, t.to_f]'
2016-02-18T11:22:11+00:00
1455794531.0707562
```